src/shared/gitlab-client.ts

```diff
@@ -149,4 +159,77 @@ export class GitlabClient {
     };
   }
 
+  public async getUserByEmail(email: string) {
+    if (!validateEmail(email)) {
+      throw Boom.badRequest(`Invalid email '${email}'`);
+    }
+    const search = {
+      search: email,
+    };
+    const users = await this.fetchJson<User[]>(`users?${qs.stringify(search)}`, true);
+    if (!users || !users.length) {
+      throw Boom.badRequest(`Can\'t find user '${email}'`);
+    }
+    if (users.length > 1) {
+      // This shoud never happen
+      throw Boom.badRequest(`Found multiple users with email '${email}'`);
+    }
+    return users[0];
+  }
+
+  public createUser(
```

Comment:
Sure. I originally had these in `authentication-hapi-plugin` so that they took the `GitlabClient` as an argument. I then realized that these only contain information about the GitLab API, i.e. the path and the required parameters. These are in some sense 'second order' API functions, but I feel that they naturally belong to `GitlabClient`. One could actually argue that `GitlabClient` wasn't much of a client before, since it didn't contain any information about the API endpoints.